containers/Dockerfile.buildenv

```diff
@@ -72,46 +76,59 @@ RUN apt-get update -y \
     && printf '. /etc/profile.d/lmod.sh\n' >> /root/.bashrc \
     && . /etc/profile.d/lmod.sh
 
-# Install Spack to get remaining dependencies
-WORKDIR /work
-RUN git clone https://github.com/LLNL/spack.git
-WORKDIR /work/spack
-# Since spack/develop is rather unstable, we check out a commit we
-# know is stable. This should be updated periodically to update
-# installed packages.
-RUN git checkout 470a45c51659156e7d154ea890e798ce32b8767d
-WORKDIR /work
-
-# Spack needs to be pointed to the system OpenSSL to work properly, we add this
-# in the general configure script for Spack rather than a user-specific
-# configure script. The below code is documented in the Spack manual.
-RUN printf "\n  openssl:\n    paths:\n      openssl@1.0.2g: /usr\n\
-    buildable: False\n" >> /work/spack/etc/spack/defaults/packages.yaml
-
-# Add Spack to PATH and install required dependencies
-# The sed commands are necessary because spack fails to find the
-# fortran compilers for an unknown reason. We compile the libraries with GCC6
-# so that we do not need separate versions for GCC6 and GCC7.
-RUN echo "export PATH=\$PATH:/work/spack/bin" >> /root/.bashrc\
-    && echo '. /work/spack/share/spack/setup-env.sh' >> /root/.bashrc \
-    && export PATH=$PATH:/work/spack/bin \
-    && spack compiler find \
-    && sed -i 's@fc: null@fc: /usr/bin/gfortran@' \
-    /root/.spack/linux/compilers.yaml \
-    && sed -i 's@f77: null@f77: /usr/bin/gfortran@' \
-    /root/.spack/linux/compilers.yaml
-RUN /work/spack/bin/spack install cmake \
-    && /work/spack/bin/spack install --no-checksum catch@2.1.0 \
-    && /work/spack/bin/spack install brigand@master \
-    && /work/spack/bin/spack install blaze \
-    && /work/spack/bin/spack install gsl%gcc@6.5.0 \
```

Comment:
For my own clarification: Was this previously redundant, or was there something about versions that needed this? It looks like we already had it in the apt.

Reply:
If memory serves me correctly then way back when this started (two years ago) the LTS ubuntu was 14.04 and so it didn't have any newer versions of these packages, or even things like catch at all. Spack was suggested by Erik Schnetter (and generally works well), but what it does is it compiles e.g. CMake from scratch for every compiler version unless you specify really long modifier strings for each package giving each dependency and exactly what compiler that dependency should be built with. While good for HPC systems, that seemed like overkill here. Compiling CMake takes something like 10 minutes, and compiling it 2-3 times meant this process took a very long time. 

If you're referring to `gsl` specifically, I think this was merely an oversight when the apt installation was added (presumably my fault :) )